src/custom/components/AccountDetails/Transaction/index.tsx

```diff
@@ -98,78 +106,99 @@ function ActivitySummary(params: {
   if (!activityData) return null
 
   const { activity, type, summary } = activityData
+  const isOrder = activity && type === ActivityType.ORDER
 
   // Order Summary default object
-  const orderSummary: OrderSummaryType = {
-    from: '🤔',
-    to: '🤔',
-    limitPrice: '🤔',
-    validTo: '🤔',
-  }
-
-  const isOrder = type === ActivityType.ORDER
-
+  let orderSummary: OrderSummaryType
   if (isOrder) {
-    const { inputToken, sellAmount, feeAmount, outputToken, buyAmount, validTo, kind, fulfillmentTime } =
-      activity as Order
+    const order = activity as Order
+    const { inputToken, sellAmount, feeAmount, outputToken, buyAmount, validTo, kind, fulfillmentTime } = order
 
     const sellAmt = CurrencyAmount.fromRawAmount(inputToken, sellAmount.toString())
     const feeAmt = CurrencyAmount.fromRawAmount(inputToken, feeAmount.toString())
     const outputAmount = CurrencyAmount.fromRawAmount(outputToken, buyAmount.toString())
-    const limitPrice = formatSmart(getLimitPrice(activity as Order))
-    // const executedPrice = formatSmart(getExecutedPrice(activity as Order))
-    const executedPrice = null
-
-    orderSummary.from = `${formatSmart(sellAmt.add(feeAmt))} ${sellAmt.currency.symbol}`
-    orderSummary.to = `${formatSmart(outputAmount)} ${outputAmount.currency.symbol}`
-    orderSummary.limitPrice = `${limitPrice} ${outputAmount.currency.symbol} per ${sellAmt.currency.symbol}`
-    orderSummary.executedPrice = executedPrice
-      ? `${executedPrice} ${outputAmount.currency.symbol} per ${sellAmt.currency.symbol}`
-      : undefined
-    orderSummary.validTo = new Date((validTo as number) * 1000).toLocaleString()
-    orderSummary.fulfillmentTime = fulfillmentTime ? new Date(fulfillmentTime).toLocaleString() : undefined
-    orderSummary.kind = kind.toString()
+    const buyTokenDecimals = order?.inputToken?.decimals ?? DEFAULT_PRECISION
+    const sellTokenDecimals = order?.inputToken?.decimals ?? DEFAULT_PRECISION
+
+    const limitPrice = formatSmart(
+      getLimitPrice({
+        buyAmount: order.buyAmount.toString(),
+        sellAmount: order.sellAmount.toString(),
+        buyTokenDecimals,
+        sellTokenDecimals,
+        inverted: false, // TODO: handle invert price
```

Comment:
My understanding is that we agreed to not allow price inversion here

Reply:
Maybe in the future we will. E.g. have an option somewhere in the sidebar to inverted all prices. Would it be ok to leave it in there, assuming the default is `false` for now?

Reply:
Yep, i'd be fine if it's not included in the MVP, but nice to leave the placeholder and include it later